### PR TITLE
qm/refactor-requires-string-to-errors

### PR DESCRIPTION
**Pull request by shababali**
**Context**
I was initially tasked w/ refactoring the requires error strings. However, Peter was on a roll and knocked it out of the park. This is a small post-implementation quality management commit. The least I could do, given his efforts! Not enough claps.

Note: Feel free to discard, but consider the one duplication

### CODE REVIEW

File: contracts/NFTOpt.sol [hunk 1]
```diff
@@ -38,26 +42,33 @@ contract NFTOpt {
...
     /// @notice Insufficient funds in escrow to withdrawal
     error INSUFFICIENT_FUNDS();
 
     /// @notice Failed to transfer funds from escrow to msg.sender
     error FUNDS_TRANSFER_FAILED();
 
-    /// @notice Buyer always != seller
-    error CANNOT_HAVE_BUYER_SAME_AS_SELLER();
 
+    // Arithmetic Error
```
Comment (Stefaanna) [inline contracts/NFTOpt.sol:68]:
```suggestion
    /// @notice Arithmetic Error

```

Comment (Stefaanna) [reply contracts/NFTOpt.sol:68]: I see the other comments as redundant, but still, we should keep it consistent and comments should have @notice before the description

Comment (shababali) [reply contracts/NFTOpt.sol:68]: sounds good, will add.
curious as to how those macros render, could we get a quick review in standup tomorrow?

Comment (grasshopper47) [reply contracts/NFTOpt.sol]: > sounds good, will add. curious as to how those macros render, could we get a quick review in standup tomorrow?

have a read here: https://blog.soliditylang.org/2021/04/21/custom-errors/

File: .gitignore [hunk 2]
```diff
@@ -1,6 +1,15 @@
...
 cache
+
+# Node/NPM
 node_modules
 .next
 typechain-types
 package-lock.json
+
+# OSX
+.DS_Store
```
Comment (grasshopper47) [inline .gitignore:12]: 👍

File: .gitignore [hunk 3]
```diff
@@ -1,6 +1,15 @@
...
+
+# IntelliJ IDE
+.idea
```
Comment (grasshopper47) [inline .gitignore:15]: we use vscode :)

**Review by grasshopper47 — APPROVED** [review]
Comments need improving; will update myself, then discuss explain in daily

Good job spotting out the duplicate condition!

File: contracts/NFTOpt.sol (hunk 1 above)
Comment (shababali) [reply contracts/NFTOpt.sol]: *^ thank you!

Comment (shababali) [reply contracts/NFTOpt.sol]: *^ thank you!

**Pull request merged**